src/traffic_cop/traffic_cop.cpp

```diff
@@ -406,6 +418,17 @@ TrafficCopResult TrafficCop::RunExecutableQuery(const common::ManagedPointer<net
     return {ResultType::ERROR, error};
   }
 
+  auto &resource_metrics = common::thread_context.resource_tracker_.GetMetrics();
+
+  const bool query_trace_metrics_enabled =
+      common::thread_context.metrics_store_ != nullptr &&
+      common::thread_context.metrics_store_->ComponentToRecord(metrics::MetricsComponent::QUERY_TRACE);
+
+  if (query_trace_metrics_enabled) {
+    common::thread_context.metrics_store_->RecordQueryTrace(exec_query->GetQueryId(), metrics::MetricsUtil::Now(),
+                                                            resource_metrics);
```

Comment:
Similarly, we don't need the `resource_metrics` here as well.

Reply:
Removed `resource_metrics`.